packages/runtime/src/templates/edge/next-dev.js

```diff
@@ -36,7 +36,9 @@ const handler = async (req, context) => {
     // We need to cache-bust the import because otherwise it will claim it
     // doesn't exist if the user creates it after the server starts
     const nextMiddleware = await import(`../../middleware.js#${++idx}`)
-    middleware = nextMiddleware.middleware
+
+    // The middleware file can export a named `middleware` export or a `default` export
+    middleware = isFunction(nextMiddleware.middleware) ? nextMiddleware.middleware : nextMiddleware.default
```

Comment:
Is the function check necessary? I imagine `nextMiddleware.middleware` is `undefined` if the `default` exists?

```suggestion
    middleware = nextMiddleware.middleware ?? nextMiddleware.default
```